app/services/idempotency/store.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
```

Comment:
I think we might need to add general error handling when talking to Redis for instance when there are connection errors etc.

Reply:
Hi @cwrw I have tested the behaviour when Redis is unavailable: the Rails.cache object simply returns nil if it is unable to connect. So everything will continue to work, just we will not be able to detect idempotency conflicts. That sounds like a pragmatic behaviour?

```
[6] pry(main)> Rails.cache.read('foo')
=> "bar"
# (turn off redis)
[9] pry(main)> Rails.cache.read('foo')
=> nil
# (turn on redis)
[10] pry(main)> Rails.cache.read('foo')
=> "bar"
```

Reply:
There is a separate piece of work to implement a Redis health check, see https://github.com/ministryofjustice/hmpps-book-secure-move-api/pull/343